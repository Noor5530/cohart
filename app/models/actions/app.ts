import { Posts, YourConnections } from '../types';

export interface GetAllPostResponse {
  type: string;
  response: Posts[];
}
export interface GetYourConnectionsResponse {
  type: string;
  response: YourConnections[];
}
export interface GetAllPostRequest {
  type: string;
}
export interface GetSuggestedConnectionResponse {
  type: string;
  response: YourConnections[];
}
export interface GetSuggestedConnectionCategoriesResponse {
  type: string;
  response: YourConnections[];
}
export interface GetPendingRequestResponse {
  type: string;
  response: YourConnections[];
}
