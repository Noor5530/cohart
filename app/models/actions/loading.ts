export interface LoadingState {
  type: string;
}
