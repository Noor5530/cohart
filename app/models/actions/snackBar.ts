export interface EnableSnackBar {
  type: string;
  data: { message: string };
}

export interface DisableSnackBar {
  type: string;
}
