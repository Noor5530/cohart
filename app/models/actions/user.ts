import { ITag } from "models/reducers/user";

export interface IRequestOtpPayload {
  phone_number?: string;
  user_code?: string;
  user_id?: string;
  non_formatted_phone_number?: string;
  country_code: string;
  request_otp_type: string;
    registration_id?: string;
    resend_sms?: boolean
}

export interface IRequestOtpState {
  type: string;
  data: IRequestOtpPayload;
}

export interface IRequestOtpResponseState {
  data: {
    data: any;
    statusCode: number;
  };
}
export interface IGetUserInformationState {
  type: string;
  data: { user_id: string | number };
}

export interface ILoginRequestState {
  type: string;
  data: { phone_number: string | number };
}

export interface IResponse {
  full_name?: string;
  location?: string;
  phone_number?: number | string;
  token?: string;
  interests?: string;
  cover_image?: string;
  subtitle?: string;
  badge?: string;
  contactInfo?: string;
  meet_me_at?: string;
  last_spotted?: string;
  currently?: string;
  affiliations?: string;
  bio?: string;
  user_interests?: string;
  tags: string[];
  accepted_guideline: boolean;
  loginAs?: boolean;
  seen_me_at?: string,
  user_moods: ITag[],
  user_mediums: ITag[],
  user_tags: ITag[]
}

export interface ILoginResponseState {
  type: string;
  response: IResponse;
}

export interface ProfileState {
  full_name: string;
  location: string;
  phone_number?: number | string;
  token?: string;
  interests?: string[];
  cover_image?: string;
  subtitle?: string;
  badge?: string;
  contactInfo?: string;
  meet_me_at?: string[];
  last_spotted?: string;
  currently?: string;
  affiliations?: string;
  bio?: string;
  title_stamp: number;
  twitter: string;
  website: string;
  instagram: string;
  post_view: boolean;
  bio_view: boolean;
  affliation_view: boolean;
  portfolio_view: boolean;
  last_spotted_view: boolean;
  meet_me_at_view: boolean;
  currently_view: boolean;
  accepted_guideline: boolean;
  first_name: string;
  last_name: string;
  previous_logged_in: boolean;
  signup_complete: boolean;
  username: string;
}

export interface EditProfileRequest {
  data: EditProfileState;
  type: string;
}

export interface EditProfileResponse {
  response: EditProfileState;
  type: string;
}

export interface UserTopicsState {
  id: string | number;
  name: string;
}

export interface SaveUserTopicRequest {
  data: any;
  type: string;
}

export interface SaveUserTopicResponse {
  response: any;
  type: string;
}

export type UpdateUserInformation = Pick<
  ProfileState,
  | "location"
  | "interests"
  | "subtitle"
  | "full_name"
  | "contactInfo"
  | "title_stamp"
>;

export type UpdateProfile = Pick<
  ProfileState,
  | "affiliations"
  | "cover_image"
  | "contactInfo"
  | "currently"
  | "meet_me_at"
  | "last_spotted"
  | "bio"
  | "title_stamp"
  | "website"
  | "twitter"
  | "instagram"
  | "affliation_view"
  | "bio_view"
  | "currently_view"
  | "last_spotted_view"
  | "meet_me_at_view"
  | "portfolio_view"
  | "post_view"
>;
export type EditProfileState = Pick<
  ProfileState,
  "first_name" | "last_name" | "location" | "full_name" | "username"
>;
export interface UpdateProfileDataRequest {
  data: UpdateProfile;
  type: string;
}

export interface UpdateProfileDataResponse {
  response: UpdateProfile;
  type: string;
}
export interface UpdateUserInformationRequest {
  data: UpdateUserInformation;
  type: string;
}

export interface UpdateUserInformationResponse {
  response: UpdateUserInformation;
  type: string;
}
export interface ChangeNumberRequest {
  type: string;
  data: {
    user_id: string;
    current_phone_number: string | number;
    new_phone_number: string;
  };
}

export interface ChangeNumberResponse {
  type: string;
  response: {
    phone_number: string;
  };
}
export type ViewTutorialBody = {
  id: string;
  view_type: string;
  view_state: boolean;
};
export interface ViewTutorialRequest {
  type: string;
  data: ViewTutorialBody;
}

export interface ViewTutorialResponse {
  type: string;
}

export enum LogOutRequestEnum {
  tokenExpire = "tokenExpire",
  clickButton = "",
}
export interface LogoutRequest {
  type: string;
  from: LogOutRequestEnum;
}