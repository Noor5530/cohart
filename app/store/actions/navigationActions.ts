/*
 * Reducer actions related with navigation
 */
import NavigationService from 'navigation/NavigationService';

export function navigateToSignUp(params?: any) {
  NavigationService.navigate('SignUp', params);
}

export function navigateToOtpVerification(params: any) {
  NavigationService.navigate('CohartLoginCode', params);
}

export function navigateToHome(params: any) {
  NavigationService.navigate('Home', params);
}
export function navigateToProfile(params: any) {
  NavigationService.navigate('Profile', params);
}
export function navigateToMyProfile(params: any) {
  NavigationService.navigate('MyProfile', params);
}
export function navigateToDiscover(params: any) {
  NavigationService.navigate('Discover', params);
}
export function navigateToDiscoverFirstTime(params: any) {
  NavigationService.navigate('DiscoverFirstTime', params);
}

export function navigateToTopic(params: any) {
  NavigationService.navigate('InterestingTopic', params);
}
export function navigateToForgotPassword(params?: any) {
  NavigationService.navigate('ForgotPassword', params);
}
export function navigateToForgotSignUp(params?: any) {
  NavigationService.navigate('SignUp', params);
}
export function reset() {
  NavigationService.reset();
}

export function navigateToCommunity(params: any) {
  NavigationService.navigate('Community', params);
}

export function navigateToWelcome(params: any) {
  NavigationService.navigate('Welcome', params);
}