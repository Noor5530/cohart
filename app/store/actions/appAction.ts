import { GetAllPostRequest, GetAllPostResponse } from 'models/actions/app';
import { Posts } from 'models/types';

import * as types from './types';

export function addNewPostRequest(data: any) {
  return {
    type: types.ADD_NEW_POST_REQUEST,
    data,
  };
}

export function addNewPostResponse(response: any) {
  return {
    type: types.ADD_NEW_POST_RESPONSE,
    response,
  };
}

export function getAllPostsRequest(data: any): GetAllPostRequest {
  return {
    type: types.GET_ALL_POST_REQUEST,
    data,
  };
}

export function startDeeplinkRequest() {
  return {
    type: types.START_DEEPLINK_REQUEST
  };
}

export function finishDeeplinkRequest() {
  return {
    type: types.FINISH_DEEPLINK_REQUEST
  };
}

export function getAllPostsResponse(response: Posts[]): GetAllPostResponse {
  return {
    type: types.GET_ALL_POST_RESPONSE,
    response,
  };
}

export function getUserPostsRequest(): GetAllPostRequest {
  return {
    type: types.GET_USER_POST_REQUEST,
  };
}

export function getUserPostsResponse(response: Posts[]): GetAllPostResponse {
  return {
    type: types.GET_USER_POST_RESPONSE,
    response,
  };
}
export function getYourConnectionRequest(data: any) {
  return {
    type: types.GET_YOUR_CONNECTIONS_REQUEST,
    data,
  };
}

export function getYourConnectionResponse(response: any) {
  return {
    type: types.GET_YOUR_CONNECTIONS_RESPONSE,
    response,
  };
}
export function getSuggestedConnectionsRequest(data: any) {
  return {
    type: types.GET_SUGGESTED_CONNECTIONS_REQUEST,
    data,
  };
}

export function getSuggestedConnectionsResponse(response: any) {
  return {
    type: types.GET_SUGGESTED_CONNECTIONS_RESPONSE,
    response,
  };
}
export function getPendingRequestRequest(data: any) {
  return {
    type: types.GET_PENDING_REQUEST_REQUEST,
    data,
  };
}

export function getPendingRequestResponse(response: any) {
  return {
    type: types.GET_PENDING_REQUEST_RESPONSE,
    response,
  };
}

export function getSuggestionConnectionCategoriesRequest(data: any) {
  return {
    type: types.GET_SUGGESTION_CATEGORIES_REQUEST,
    data,
  };
}

export function getSuggestionConnectionCategoriesResponse(response: any) {
  return {
    type: types.GET_SUGGESTION_CATEGORIES_RESPONSE,
    response,
  };
}

export function fetchUserWorkRequest() {
  return {
    type: types.FETCH_USER_WORK_REQUEST,
  };
}

export function fetchUserWorkResponse(response: any) {
  return {
    type: types.FETCH_USER_WORK_RESPONSE,
    response,
  };
}

export function getAcceptPendingRequest(data: any) {
  return {
    type: types.GET_ACCEPT_PENDING_REQUEST_REQUEST,
    data,
  };
}

export function rejectPendingRequest(data: any) {
  return {
    type: types.REJECT_PENDING_REQUEST_REQUEST,
    data,
  };
}

export function addNewGroupRequest(data: any) {
  return {
    type: types.ADD_GROUP_REQUEST,
    data,
  };
}

export function addNewGroupResponse(response: any) {
  return {
    type: types.ADD_GROUP_REQUEST,
    response,
  };
}

export function addUserInGroupRequest(data: any) {
  return {
    type: types.ADD_GROUP_USER_REQUEST,
    data,
  };
}

export function addUserInGroupResponse(response: any) {
  return {
    type: types.ADD_GROUP_USER_RESPONSE,
    response,
  };
}

export function getAllUserGroupRequest(data: any) {
  return {
    type: types.FETCH_USER_GROUP_REQUEST,
    data,
  };
}

export function getAllUserGroupResponse(response: any) {
  return {
    type: types.FETCH_GROUP_USER_RESPONSE,
    response,
  };
}

export function getSavedWorkRequest(data: object) {
  return {
    type: types.GET_MY_WORK_REQUEST,
    data,
  };
}

export function getSavedWorkResponse(response: object) {
  return {
    type: types.GET_MY_WORK_RESPONSE,
    response,
  };
}

export function logOut() {
  return {
    type: types.LOG_OUT,
  };
}

export function socketEvent(response: any) {
  return {
    type: types.TOGGLE_SOCKET_EVENT,
    response,
  };
}
