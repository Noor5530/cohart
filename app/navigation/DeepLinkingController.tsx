import React from 'react';
import {  View } from 'react-native';
import branch from 'react-native-branch';

import * as navigationActions from 'store/actions/navigationActions';


export interface IDeepLinkingControllerProps {
    isLoggedIn: boolean;
    userId: string | number | undefined;
    isFirstTimeLogin: boolean,
    onProcessDeeplink: () => void;
    username: string 
  }

export default class DeepLinkingController extends React.Component<IDeepLinkingControllerProps> {

    componentDidMount() {
        branch.subscribe(this.onHandleDeepLink);
    }

    onHandleDeepLink = ({error, params} : any) => {
            if (error) {
              return
            }            
            const { postId, deepLinkType, username } = params;
            
            const { isLoggedIn, onProcessDeeplink, username: userNameProps } = this.props;

            if(deepLinkType === 'ArtistProfile'){
                if(isLoggedIn) {
                    const currentUser = userNameProps === username
                    navigationActions.navigateToMyProfile({username, currentUser, isFromDeepLink: true})
                    return
                }
                navigationActions.navigateToMyProfile({username, currentUser: false, isFromDeepLink: true})
                return
            }
            
            if (!postId) return;

            if (params && postId && postId != undefined) {
                onProcessDeeplink();
            }

            if (isLoggedIn) {
                navigationActions.navigateToDiscover({ postId });
            }

            if(!isLoggedIn) {
                navigationActions.navigateToWelcome({ postId })
            }
    }


    render() {
        return <View />;
    }
}
