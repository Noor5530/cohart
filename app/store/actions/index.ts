import * as appAction from './appAction';
import * as loadingAction from './loadingActions';
import * as navigationActions from './navigationActions';
import * as snackBarAction from './snackBarAction';
import * as userActions from './userActions';

// export action creators
export const ActionCreators = Object.assign(
  {},
  appAction,
  userActions,
  loadingAction,
  snackBarAction,
  navigationActions,
);
