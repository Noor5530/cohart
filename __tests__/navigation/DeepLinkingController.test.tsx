/**
 * @format
 */


 
import 'react-native';
import React from 'react'; 
import renderer from 'react-test-renderer';
import branch, {  } from 'react-native-branch';
import { render } from "@testing-library/react-native";
import * as navigationActions from 'store/actions/navigationActions';

import DeepLinkingController, { IDeepLinkingControllerProps } from 'navigation/DeepLinkingController';

 jest.useFakeTimers();

 
 test('renders DeepLinkingController component correctly and deeplink branch sdk should be subscribe', () => {
   const props: IDeepLinkingControllerProps = {
      isFirstTimeLogin: false,
      userId: '123',
      isLoggedIn: true,
      onProcessDeeplink: () => {}
    };
   const tree = renderer.create(<DeepLinkingController {...props} />).toJSON();
   expect(tree).toMatchSnapshot();

   expect(branch.subscribe).toBeCalled();

 });

 test('Should do nothing when post id is not recognized', () => {
  const props: IDeepLinkingControllerProps = {
     isFirstTimeLogin: false,
     userId: '123',
     isLoggedIn: true,
     onProcessDeeplink: jest.fn()
   };
    const tree = render(<DeepLinkingController {...props} />);
    const instance = tree.UNSAFE_getByType(DeepLinkingController).instance;
    
    const callbackParamsWithDeepLink = {
      error: null,
      params: {
        postId: null
      }
    }
    instance.onHandleDeepLink(callbackParamsWithDeepLink);
    expect(props.onProcessDeeplink).not.toBeCalled();

});

test('Should do nothing when receive the error with deeplink', () => {
  const props: IDeepLinkingControllerProps = {
     isFirstTimeLogin: false,
     userId: '123',
     isLoggedIn: true,
     onProcessDeeplink: jest.fn()
   };
    const tree = render(<DeepLinkingController {...props} />);
    const instance = tree.UNSAFE_getByType(DeepLinkingController).instance;
    
    const callbackParamsWithDeepLink = {
      error: 'ErrorDeeplink-URI is not valid',
      params: {
        postId: null
      }
    }
    instance.onHandleDeepLink(callbackParamsWithDeepLink);
    expect(props.onProcessDeeplink).not.toBeCalled();

});

test('Should navigate to welcome page if user is not login yet', () => {
  const props: IDeepLinkingControllerProps = {
     isFirstTimeLogin: false,
     userId: '123',
     isLoggedIn: false,
     onProcessDeeplink: jest.fn()
   };
    const tree = render(<DeepLinkingController {...props} />);
    const instance = tree.UNSAFE_getByType(DeepLinkingController).instance;
    
    const callbackParamsWithDeepLink = {
      error: null,
      params: {
        postId: 'post_id_valid'
      }
    };

    jest.spyOn(navigationActions, 'navigateToWelcome')
      .mockImplementation(() => {});

    instance.onHandleDeepLink(callbackParamsWithDeepLink);
    expect(props.onProcessDeeplink).toBeCalled();
    expect(navigationActions.navigateToWelcome).toBeCalledWith({ postId: callbackParamsWithDeepLink.params.postId })

});

test('Should navigate to discover page if user is login ', () => {
  const props: IDeepLinkingControllerProps = {
     isFirstTimeLogin: false,
     userId: '123',
     isLoggedIn: true,
     onProcessDeeplink: jest.fn()
   };
    const tree = render(<DeepLinkingController {...props} />);
    const instance = tree.UNSAFE_getByType(DeepLinkingController).instance;
    
    const callbackParamsWithDeepLink = {
      error: null,
      params: {
        postId: 'post_id_valid'
      }
    };

    jest.spyOn(navigationActions, 'navigateToDiscover')
      .mockImplementation(() => {});

    instance.onHandleDeepLink(callbackParamsWithDeepLink);
    expect(props.onProcessDeeplink).toBeCalled();
    expect(navigationActions.navigateToDiscover).toBeCalledWith({ postId: callbackParamsWithDeepLink.params.postId })

});

 
 