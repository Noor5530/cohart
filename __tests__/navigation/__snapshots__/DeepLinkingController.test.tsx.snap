// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DeepLinkingController component correctly and deeplink branch sdk should be subscribe 1`] = `<View />`;
